src/service/logs/kinesis_firehose.rs

```diff
@@ -112,71 +113,70 @@ pub async fn process(
+                        local_val.insert("logGroup".to_owned(), kfh_data.log_group.clone().into());
+                        local_val
+                            .insert("logStream".to_owned(), kfh_data.log_stream.clone().into());
+                        local_val.insert(
+                            "subscriptionFilters".to_owned(),
+                            kfh_data.subscription_filters.clone().into(),
+                        );
+
+                        let local_msg = event.message.as_str().unwrap();
+
+                        if local_msg.starts_with('{') && local_msg.ends_with('}') {
+                            let result: Result<json::Value, json::Error> =
+                                json::from_str(local_msg);
+
+                            match result {
+                                Err(_e) => {
+                                    local_val.insert("message".to_owned(), event.message.clone());
+                                }
+                                Ok(message_val) => {
+                                    local_val.insert("message".to_owned(), message_val.clone());
+                                }
                             }
+                        } else {
+                            local_val.insert("message".to_owned(), local_msg.into());
                         }
-                    } else {
-                        local_val.insert("message".to_owned(), local_msg.into());
-                    }
 
-                    value = local_val.clone().into();
-
-                    // Start row based transform
-                    #[cfg(feature = "zo_functions")]
-                    let value = crate::service::ingestion::apply_stream_transform(
-                        &local_tans,
-                        &value,
-                        &lua,
-                        &stream_lua_map,
-                        &stream_vrl_map,
-                        stream_name,
-                        &mut runtime,
-                    );
-                    #[cfg(feature = "zo_functions")]
-                    if value.is_null() || !value.is_object() {
-                        stream_status.status.failed += 1; // transform failed or dropped
-                        continue;
+                        value = local_val.clone().into();
+                        // handling of timestamp
+                        timestamp = match event.timestamp {
+                            Some(v) => match parse_i64_to_timestamp_micros(v) {
+                                Ok(t) => t,
+                                Err(e) => {
+                                    stream_status.status.failed += 1;
+                                    stream_status.status.error = e.to_string();
+                                    continue;
+                                }
+                            },
+                            None => Utc::now().timestamp_micros(),
+                        };
```

Comment:
Define a helper function
```rust
fn try_parse_timestamp(t: Option<i64>) -> Result<i64, anyhow::Error> {
    t.map_or_else(
        || Ok(Utc::now().timestamp_micros()),
        parse_i64_to_timestamp_micros,
    )
}
```
and use it here ..
```suggestion
                        timestamp = match try_parse_timestamp(event.timestamp) {
                            Ok(t) => t,
                            Err(e) => {
                                stream_status.status.failed += 1;
                                stream_status.status.error = e.to_string();
                                continue;
                            }
                        };```